url/resources/percent-encoding.json

```diff
@@ -0,0 +1,18 @@
+[
+  "Tests for percent-encoding.",
```

Comment:
I guess this is really tests for "percent-encode after encoding"? Which is why the input is a string code point, instead of a byte sequence?

What are the values used for percentEncodeSet and spaceAsPlus?

Reply:
I guess currently it's special-query percent-encode set and false, but should be easy to expand. Not sure how much time I want to put into this yet though.

Reply:
Makes sense. It's worth documenting those parameters (and the correct algorithm name) in the meantime.